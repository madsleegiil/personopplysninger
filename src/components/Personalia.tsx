import * as React from 'react';

const Personalia = () => {
    return (
        <div className="panel">
            Personalia
        </div>
    );
};

export default Personalia;
