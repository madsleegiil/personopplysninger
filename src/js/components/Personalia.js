import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Box from 'js/components/Box';

class Personalia extends Component {
  render() {
    return (
      <Box header="Personalia">
        <ul className="personalia-list">
          <li>
            <span className="title">Fornavn</span>
            <span className="content">{this.props.navn.fornavn}</span>
          </li>
          <li>
            <span className="title">Etternavn</span>
            <span className="content">{this.props.navn.slektsnavn}</span>
          </li>
          <li>
            <span className="title">Fødselsnummer</span>
            <span className="content">{this.props.ident}</span>
          </li>
          <li>
            <span className="title">Utenlandsk ID</span>
            <span className="content">x</span>
          </li>
          <li>
            <span className="title">Telefon</span>
            <span className="content">{this.props.telefon.privat}</span>
          </li>
          <li>
            <span className="title">Kontonummer</span>
            <span className="content">{this.props.kontonummer.nummer}</span>
          </li>
          <li>
            <span className="title">e-post</span>
            <span className="content">x</span>
          </li>
          <li>
            <span className="title">Språk</span>
            <span className="content">{this.props.spraak.kode}</span>
          </li>
          <li>
            <span className="title">Statsborgerskap</span>
            <span className="content">{this.props.statsborgerskap.kode}</span>
          </li>
          <li>
            <span className="title">Personstatus</span>
            <span className="content">{this.props.status.kode}</span>
          </li>
          <li>
            <span className="title">Fødested</span>
            <span className="content">x</span>
          </li>
          <li>
            <span className="title">Spesialregistrering</span>
            <span className="content">x</span>
          </li>
          <li>
            <span className="title">Sivilstand</span>
            <span className="content">{this.props.sivilstand.kode}</span>
          </li>
          <li>
            <span className="title">Sikkerhetstiltak</span>
            <span className="content">{this.props.tiltak.type}</span>
          </li>
          <li>
            <span className="title">Kjønn</span>
            <span className="content">{this.props.kjonn}</span>
          </li>
          <li>
            <span className="title">Tilpasset kommunikasjon</span>
            <span className="content">x</span>
          </li>
        </ul>
        <div className="box-footer">
          Kilde: NAV, Skatteetaten, Folkeregisteret, Kontakt- og reservasjonsregisteret
        </div>
      </Box>
    );
  }
}

Personalia.propTypes = {
  navn: PropTypes.shape.isRequired,
  ident: PropTypes.string.isRequired,
  statsborgerskap: PropTypes.shape.isRequired,
  status: PropTypes.shape.isRequired,
  telefon: PropTypes.shape.isRequired,
  tiltak: PropTypes.shape.isRequired,
  kjonn: PropTypes.string.isRequired,
  spraak: PropTypes.shape.isRequired,
  sivilstand: PropTypes.shape.isRequired,
  kontonummer: PropTypes.shape.isRequired,
};

export default Personalia;
