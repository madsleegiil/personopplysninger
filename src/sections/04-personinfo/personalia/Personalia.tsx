/* eslint-disable */

import React, { Component } from "react";
import Box from "../../../components/box/Box";
import Kilde from "../../../components/kilde/Kilde";
import kvinne from "../../../assets/img/kvinne.svg";
import mann from "../../../assets/img/mann.png";
import ListElement from "../../../components/listelement/ListElement";
import { Personalia as PersonaliaType } from "../../../types/personalia";

interface Props {
  personalia: PersonaliaType;
}

class Personalia extends Component<Props> {
  render() {
    const {
      personident,
      fornavn,
      etternavn,
      kontonr,
      spraak,
      statsborgerskap,
      foedested,
      sivilstand,
      kjoenn
    } = this.props.personalia;

    const personidentHeader =
      personident && personident.type === "DNR"
        ? "personalia.dnr"
        : "personalia.fnr";

    const fornavnHeader =
      fornavn && fornavn.indexOf(" ") === -1
        ? "personalia.first_name"
        : "personalia.first_and_middle_name";

    const formattertKontonr =
      kontonr && kontonr.length === 11
        ? kontonr.replace(/^(.{4})(.{2})(.*)$/, "$1 $2 $3")
        : kontonr;

    const formattertPersonident =
      personident && personident.verdi.length === 11
        ? {
            ...personident,
            verdi: personident.verdi.replace(/^(.{6})(.*)$/, "$1 $2")
          }
        : personident;

    return (
      <Box
        id="personalia"
        header="Personalia"
        icon={kjoenn === "Mann" ? mann : kvinne}
        infoType="personalia"
      >
        <ul className="list-column-2">
          <ListElement titleId={fornavnHeader} content={fornavn} />
          <ListElement
            className="capitalize"
            titleId="personalia.surname"
            content={etternavn}
          />
          {formattertPersonident && (
            <ListElement
              titleId={personidentHeader}
              content={formattertPersonident.verdi}
            />
          )}
          <ListElement titleId="personalia.language" content={spraak} />
          <ListElement
            titleId="personalia.citizenship"
            content={statsborgerskap}
          />
          <ListElement titleId="personalia.birthplace" content={foedested} />
          <ListElement titleId="personalia.civil_status" content={sivilstand} />
          <ListElement titleId="personalia.gender" content={kjoenn} />
        </ul>
        <Kilde
          tekst="personalia.source"
          lenkeTekst="personalia.link.folkeregisteret"
          href="https://www.skatteetaten.no/person/folkeregister/"
        />
        <hr className="box__linje-bred" />
        <ul className="list-column-2">
          <ListElement
            titleId="personalia.account_no"
            content={formattertKontonr}
          />
        </ul>
      </Box>
    );
  }
}

export default Personalia;
