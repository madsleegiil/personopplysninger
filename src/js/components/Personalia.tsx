/* eslint-disable */

import React, { Component } from "react";
import { FormattedMessage } from "react-intl";
import { Undertekst } from "nav-frontend-typografi";
import Box from "./Box";
import kvinne from "../../assets/img/kvinne.svg";
import mann from "../../assets/img/mann.png";
import ListElement from "./ListElement";
import { Personalia as PersonaliaType } from "../../types/personalia";

class Personalia extends Component<PersonaliaType> {
  render() {
    const {
      personident,
      fornavn,
      etternavn,
      kontonr,
      spraak,
      statsborgerskap,
      foedested,
      sivilstand,
      kjoenn
    } = this.props;

    const personidentHeader =
      personident && personident.type === "DNR"
        ? "personalia.dnr"
        : "personalia.fnr";

    const fornavnHeader =
      fornavn && fornavn.indexOf(" ") === -1
        ? "personalia.first_name"
        : "personalia.first_and_middle_name";

    return (
      <Box
        id="personalia"
        header="Personalia"
        icon={kjoenn === "Mann" ? mann : kvinne}
        infoType="personalia"
      >
        <ul className="list-column-2">
          {fornavn && <ListElement titleId={fornavnHeader} content={fornavn} />}
          {etternavn && (
            <ListElement
              classNameContent="capitalize"
              titleId="personalia.surname"
              content={etternavn}
            />
          )}
          {personident && personident.verdi && (
            <ListElement
              titleId={personidentHeader}
              content={personident.verdi}
            />
          )}
          {kontonr && (
            <ListElement titleId="personalia.account_no" content={kontonr} />
          )}
          {spraak && (
            <ListElement titleId="personalia.language" content={spraak} />
          )}
          {statsborgerskap && (
            <ListElement
              titleId="personalia.citizenship"
              content={statsborgerskap}
            />
          )}
          {foedested && (
            <ListElement titleId="personalia.birthplace" content={foedested} />
          )}
          {sivilstand && (
            <ListElement
              titleId="personalia.civil_status"
              content={sivilstand}
            />
          )}
          {kjoenn && (
            <ListElement titleId="personalia.gender" content={kjoenn} />
          )}
        </ul>
        <div className="box-footer">
          <Undertekst>
            <FormattedMessage id="personalia.source" />
          </Undertekst>
        </div>
      </Box>
    );
  }
}

export default Personalia;
