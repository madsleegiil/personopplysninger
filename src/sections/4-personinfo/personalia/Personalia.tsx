import React, { Component } from "react";
import Box from "../../../components/box/Box";
import kvinne from "../../../assets/img/kvinne.svg";
import mann from "../../../assets/img/mann.png";
import Kontonummer from "./subsections/Kontonummer";
import Folkeregisteret from "./subsections/Folkeregisteret";
import Telefonnummer from "./subsections/Telefonnummer";
import DKIF from "./subsections/DKIF";
import { Personalia as PersonaliaType } from "../../../types/personalia";

interface Props {
  personalia: PersonaliaType;
}

class Personalia extends Component<Props> {
  render() {
    const { kontonr, kjoenn } = this.props.personalia;

    return (
      <Box
        id="personalia"
        header="Personalia"
        icon={kjoenn === "Mann" ? mann : kvinne}
        infoType="personalia"
      >
        <hr className="box__linje-bred" />
        <Folkeregisteret personalia={this.props.personalia} />
        <Kontonummer kontonummer={kontonr} />
        <Telefonnummer tlfnr={this.props.personalia.tlfnr} />
        <DKIF />
      </Box>
    );
  }
}

export default Personalia;
