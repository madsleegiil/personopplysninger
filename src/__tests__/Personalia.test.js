import * as React from 'react';
import wrapIntl from 'js/IntlTestHelper';
import Personalia from 'js/components/Personalia';
const ReactTestRenderer = require('react-test-renderer');

test('render Personalia with content', () => {
  const personalia = {
    fornavn: '',
    etternavn: '',
    personident: {
      verdi: '',
      type: '',
    },
    kontonr: '',
    tlfnr: {
      jobb: '',
      mobil: '',
      privat: '',
    },
    spraak: '',
    epostadr: '',
    statsborgerskap: '',
    foedested: '',
    sivilstand: '',
    kjoenn: '',
  };

  const component = ReactTestRenderer.create(wrapIntl(<Personalia
    fornavn={personalia.fornavn}
    etternavn={personalia.etternavn}
    personident={personalia.personident}
    kontonr={personalia.kontonr}
    tlfnr={personalia.tlfnr}
    spraak={personalia.spraak}
    epostadr={personalia.epostadr}
    statsborgerskap={personalia.statsborgerskap}
    foedested={personalia.foedested}
    sivilstand={personalia.sivilstand}
    kjoenn={personalia.kjoenn}
  />));
  expect(component.toJSON()).toMatchSnapshot();
});
