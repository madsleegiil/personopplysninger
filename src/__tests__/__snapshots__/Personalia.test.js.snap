// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Personalia with content 1`] = `
<div
  className="BoxContainer"
>
  <div
    className="BoxWithHeader"
  >
    <h1>
      Personalia
    </h1>
    <div
      className="Box"
    >
      <ul
        className="personalia-list"
      >
        <li>
          <span
            className="title"
          >
            Fornavn
          </span>
          <span
            className="content"
          >
            Ole
          </span>
        </li>
        <li>
          <span
            className="title"
          >
            Etternavn
          </span>
          <span
            className="content"
          >
            Nordmann
          </span>
        </li>
        <li>
          <span
            className="title"
          >
            Fødselsnummer
          </span>
          <span
            className="content"
          >
            08048503978
          </span>
        </li>
        <li>
          <span
            className="title"
          >
            Utenlandsk ID
          </span>
          <span
            className="content"
          >
            x
          </span>
        </li>
        <li>
          <span
            className="title"
          >
            Telefon
          </span>
          <span
            className="content"
          >
            51515151
          </span>
        </li>
        <li>
          <span
            className="title"
          >
            Kontonummer
          </span>
          <span
            className="content"
          />
        </li>
        <li>
          <span
            className="title"
          >
            e-post
          </span>
          <span
            className="content"
          >
            x
          </span>
        </li>
        <li>
          <span
            className="title"
          >
            Språk
          </span>
          <span
            className="content"
          >
            Norsk
          </span>
        </li>
        <li>
          <span
            className="title"
          >
            Statsborgerskap
          </span>
          <span
            className="content"
          >
            Norsk
          </span>
        </li>
        <li>
          <span
            className="title"
          >
            Personstatus
          </span>
          <span
            className="content"
          >
            
          </span>
        </li>
        <li>
          <span
            className="title"
          >
            Fødested
          </span>
          <span
            className="content"
          >
            x
          </span>
        </li>
        <li>
          <span
            className="title"
          >
            Spesialregistrering
          </span>
          <span
            className="content"
          >
            12345678911
          </span>
        </li>
        <li>
          <span
            className="title"
          >
            Sivilstand
          </span>
          <span
            className="content"
          >
            Gift
          </span>
        </li>
        <li>
          <span
            className="title"
          >
            Sikkerhetstiltak
          </span>
          <span
            className="content"
          >
            Kun telefonkontakt
          </span>
        </li>
        <li>
          <span
            className="title"
          >
            Kjønn
          </span>
          <span
            className="content"
          >
            Mann
          </span>
        </li>
        <li>
          <span
            className="title"
          >
            Tilpasset kommunikasjon
          </span>
          <span
            className="content"
          >
            x
          </span>
        </li>
      </ul>
      <div
        className="box-footer"
      >
        Kilde: NAV, Skatteetaten, Folkeregisteret, Kontakt- og reservasjonsregisteret
      </div>
    </div>
  </div>
</div>
`;
